### ISSUE TITLE
Validation: CSS classes?

### ISSUE THREAD

**Issue by blissi** [MrWolfZ/ngrx-forms#34]
Hallo,
from Angular ReactiveForms I'm used to it that the CSS classes are set on the form controls.
Can this be enabled with this library somehow? Is it an out-of-the-box feature that I have overlooked?

**Comment by MrWolfZ** [comment]
Hmm, I wasn't aware that reactive forms sets classes on elements. Can you point me to the documentation or some examples? In general I am not sure though if I would like to introduce this feature. If at all I'll make it optional. While I am thinking about this you could use the following custom directive to get what you want (I type these things up directly in the browser without even compiling so I can't guarantee that it will work, but it should provide you with a starting point):

```typescript
@Directive({
  selector: '[ngrxFormControlState]',
})
export class SetFormControlClassDirective {
  @Input() set ngrxFormControlState(state: FormControlState<any>) {
    // put your logic here to set the classes whenever the state changes, e.g.
    if (state.isInvalid) {
      this.renderer.addClass(this.element.nativeElement, 'ngrx-forms-invalid');
    } else {
      this.renderer.removeClass(this.element.nativeElement, 'ngrx-forms-invalid');
    }
  };

  constructor(
    private element: Element,
    private renderer: Renderer2,
  ) {}
}
```

**Comment by blissi** [comment]
Thanks, I will try it out.
Regarding your question, see here: https://angular.io/guide/form-validation#control-status-css-classes

**Comment by MrWolfZ** [comment]
I'll leave this issue open to track this feature request.

**Comment by blissi** [comment]
Here is my implementation of it:
```
import {Directive, ElementRef, Input, Renderer2} from "@angular/core";
import {FormControlState} from "ngrx-forms";


/**
 * This directive automatically adds the Angular CSS classes to ngrx-forms controls.
 * See
 * https://angular.io/guide/form-validation#control-status-css-classes
 * and
 * https://github.com/MrWolfZ/ngrx-forms/issues/34#issuecomment-358785776
 */
@Directive({
	selector: "[ngrxFormControlState]",
})
export class NgrxSetFormControlClassDirective {
	@Input()
	set ngrxFormControlState(state: FormControlState<any>) {
		this.addOrRemoveCssClass(state.isValid, "ng-valid");
		this.addOrRemoveCssClass(state.isInvalid, "ng-invalid");
		this.addOrRemoveCssClass(state.isValidationPending, "ng-pending");
		this.addOrRemoveCssClass(state.isPristine, "ng-pristine");
		this.addOrRemoveCssClass(state.isDirty, "ng-dirty");
		this.addOrRemoveCssClass(state.isUntouched, "ng-untouched");
		this.addOrRemoveCssClass(state.isTouched, "ng-touched");
	};
	
	constructor(private element: ElementRef, private renderer: Renderer2,) {
	}
	
	private addOrRemoveCssClass(add: boolean, cssClass: string): void {
		if (add) {
			this.renderer.addClass(this.element.nativeElement, cssClass);
		} else {
			this.renderer.removeClass(this.element.nativeElement, cssClass);
		}
	}
}
```

**Comment by MrWolfZ** [comment]
I decided to include this feature unconditionally after all. I have just released version 2.3.0 which contains this feature. See the [documentation](https://github.com/MrWolfZ/ngrx-forms/blob/master/docs/FORM_CONTROLS.md#status-css-classes) for more details.
